xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -477,6 +478,59 @@ bool CInputStreamAddon::IsRealtime()
   return false;
 }
 
+
+// IChapter
+CDVDInputStream::IChapter* CInputStreamAddon::GetIChapter()
+{
+  if ((m_caps.m_mask & INPUTSTREAM_CAPABILITIES::SUPPORTS_ICHAPTER) == 0)
+    return nullptr;
+
+  return this;
+}
+
+int CInputStreamAddon::GetChapter()
+{
+  if (m_struct.toAddon.get_chapter)
+    return m_struct.toAddon.get_chapter(&m_struct);
+
+  return -1;
+}
+
+int CInputStreamAddon::GetChapterCount()
+{
+  if (m_struct.toAddon.get_chapter_count)
+    return m_struct.toAddon.get_chapter_count(&m_struct);
+
+  return 0;
+}
+
+void CInputStreamAddon::GetChapterName(std::string& name, int ch)
```

Comment:
Only a question. Is this `std::string& name` value always a empty string on his call? Think yes (seen this two places below where it becomes called).
Not that it become in conflict if `std::string` where never should give `nullptr` for the `if (res)` check (but this check is always good and more safe :smile:).